### ISSUE TITLE
Fix failing SMC tests

### ISSUE THREAD

**Issue by wleoncio** [ocbe-uio/BayesMallows#74]
Some tests on `test_smc_individual_functions.R` are failing due to not matching their expectations.

**Comment by wleoncio** [comment]
Hey @anjastein, maybe you can help me sort this out.

Consider the following version of the package:

```r
remotes::install_github("ocbe-uio/BayesMallows", "52a405")
library("BayesMallows")
```

Also, please consider the test script [here](https://gist.github.com/wleoncio/180c1430328998cb43cb962d72ee1e87#file-test-metropolis_hastings_rho-r) (based on the files you've sent me earlier).

When I generate `test_3` (L37), I get a different ranking on my machine (see image below):

![image](https://user-images.githubusercontent.com/8234768/114045140-18900d00-9888-11eb-9a8e-ab244c14c1eb.png)

Line 41 is also a bit problematic because it says `test_1` but I understand it should be `test_3`. I get the rank distance ¨0" when running `BayesMallows:::get_rank_distance(rho, test_3, metric= "ulam")`. My two questions here:

1. Can you reproduce my observations given the code above?
2. Can you reproduce the outcome expected by the [test script](https://gist.github.com/wleoncio/180c1430328998cb43cb962d72ee1e87#file-test-metropolis_hastings_rho-r) using the same conditions from back then (your machine, the original SMC-Mallows code and the published version of BayesMallows (from `install.packages("BayesMallows")`, i.e. 1.0.1))? There's a chance the output on the script is a typo.

**Comment by wleoncio** [comment]
BTW, I've fixed 3 of the failing tests. There was an obvious typo on the lines below, that had `test_2` instead of `test_3` (stupid copy and paste 🤦🏽‍♂️).

https://github.com/ocbe-uio/BayesMallows/blob/52a405cc926a287ea79719e2829c8be9e2d27b7d/tests/testthat/test_smc_individual_functions.R#L100-L103

Anyway, this is already fixed locally and will be part of the PR closing this issue.

**Comment by anjastein** [comment]
Hi Waldir,

Using my original test scripts and the correction (test 3 instead of test 1) I do get (1,2,3,4,5,6) and a rank distance of 0. All other outputs remain the same in my original test scripts.

I had a lot of trouble getting all updated versions of R packages to load on my monitor hence why I have taken so long to get back to you.

Get_mallows_loglik passes the test.

Metropolis_hastings_rho gives me this failure message

***@***.***D72CA8.2A1FBD50]


Leap_and_shift is now giving me these failure messages.
***@***.***D72CA8.2A1FBD50]


I shall keep playing with the code to see what happens.

Anja

From: Waldir Leoncio ***@***.***>
Sent: 08 April 2021 15:42
To: ocbe-uio/BayesMallows ***@***.***>
Cc: Stein, Anja (Student) ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [ocbe-uio/BayesMallows] Fix failing SMC tests (#74)


This email originated outside the University. Check before clicking links or attachments.

Hey @anjastein<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fanjastein&data=04%7C01%7Ca.stein1%40lancaster.ac.uk%7C71abf5cfb411422ec74308d8fa9c87a5%7C9c9bcd11977a4e9ca9a0bc734090164a%7C0%7C0%7C637534897490007580%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=xzwC%2Bca6KHIReOqLHR3FZlqx0J4FHdsoTotCuhMWb2c%3D&reserved=0>, maybe you can help me sort this out.

Consider the following version of the package:

remotes::install_github("ocbe-uio/BayesMallows", "52a405")



library("BayesMallows")

Also, please consider the test script here<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgist.github.com%2Fwleoncio%2F180c1430328998cb43cb962d72ee1e87%23file-test-metropolis_hastings_rho-r&data=04%7C01%7Ca.stein1%40lancaster.ac.uk%7C71abf5cfb411422ec74308d8fa9c87a5%7C9c9bcd11977a4e9ca9a0bc734090164a%7C0%7C0%7C637534897490007580%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=jtwRBWX9DFVDo3phittrFoKhNcB4PBaBud%2BxibtR7zM%3D&reserved=0> (based on the files you've sent me earlier).

When I generate test_3 (L37), I get a different ranking on my machine (see image below):

[image]<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F8234768%2F114045140-18900d00-9888-11eb-9a8e-ab244c14c1eb.png&data=04%7C01%7Ca.stein1%40lancaster.ac.uk%7C71abf5cfb411422ec74308d8fa9c87a5%7C9c9bcd11977a4e9ca9a0bc734090164a%7C0%7C0%7C637534897490007580%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=NxTlrScAY1z0KEOqCmAuht5U2Zu8WK7C2rK58WCHbVw%3D&reserved=0>

Line 41 is also a bit problematic because it says test_1 but I understand it should be test_3. I get the rank distance ¨0" when running BayesMallows:::get_rank_distance(rho, test_3, metric= "ulam"). My two questions here:

  1.  Can you reproduce my observations given the code above?
  2.  Can you reproduce the outcome expected by the test script<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgist.github.com%2Fwleoncio%2F180c1430328998cb43cb962d72ee1e87%23file-test-metropolis_hastings_rho-r&data=04%7C01%7Ca.stein1%40lancaster.ac.uk%7C71abf5cfb411422ec74308d8fa9c87a5%7C9c9bcd11977a4e9ca9a0bc734090164a%7C0%7C0%7C637534897490017541%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=UXdty8TFMjiZWeMW87XhGfI3zOwoK5lwYolbSb9xQ70%3D&reserved=0> using the same conditions from back then (your machine, the original SMC-Mallows code and the published version of BayesMallows (from install.packages("BayesMallows"), i.e. 1.0.1))? There's a chance the output on the script is a typo.

-

**Comment by wleoncio** [comment]
Great, thanks for checking! I'll update the expectations and push the results.
